CONTRIBUTING.md

```diff
@@ -1,102 +1,51 @@
+
+### Pull Requests
+- Big changes, changes to the API, or changes with backward compatibility trade-offs should be first discussed in the [Arduino Developers Mailing List](https://groups.google.com/a/arduino.cc/forum/#!forum/developers).
+- Search [existing pull requests](https://github.com/arduino/Arduino/pulls?q=) to see if one has already been submitted for this change. Search the [issues](https://github.com/arduino/Arduino/issues?q=is%3Aissue) to see if there has been discussion on this topic and whether your pull request can close any issues.
+- Code formatting should be consistent with the style used in the existing code.
+- Don't leave commented out code. A record of this code is already preserved in the commit history.
+- Note that the Arduino core libraries support many boards and processors. When fixing or adding functionality for one of them, it's easy to break something on the others. Please test your changes on as many processors as possible. Even if you don't have a particular board, try compiling your patch for it to make sure that you haven't introduced any errors.
+- All commits must be atomic. This means that the commit completely accomplishes a single task. Each commit should result in fully functional code. Multiple tasks should not be combined in a single commit. For more information see http://www.freshconsulting.com/atomic-commits.
+- Each pull request should address a single bug fix or enhancement. This may consist of multiple commits. If you have multiple, unrelated fixes or enhancements to contribute, submit them as separate pull requests.
+- Commit messages:
+  - Use the [imperative mood](http://chris.beams.io/posts/git-commit/#imperative) in the title. For example: "Apply editor.indent preference"
+  - Capitalize the title.
+  - Do not end the title with a period.
+  - Separate title from body with a blank line. If you're committing via GitHub or GitHub Desktop this will be done automatically.
+  - Wrap body at 72 characters.
+  - Completely explain the purpose of the commit. Include rationale for the change, any caveats, side-effects, etc.
+  - If your pull request fixes an issue in the issue tracker, use the [closes/fixes/resolves syntax](https://help.github.com/articles/closing-issues-via-commit-messages) in the body to indicate this.
+  - See http://chris.beams.io/posts/git-commit for more tips on writing good commit messages.
+- Rebasing pull requests is OK and encouraged. After submitting your pull request some changes may be requested. Rather than adding unnecessary extra commits to the pull request you can squash these changes into the existing commit and then do a force push to your fork. Leave a comment on the pull request thread to explain that the history has been changed. This will help to keep the commit history of the repository clean.
```

Comment:
Perhaps add: When you do a force push to your fork, the PR will be updated with your new changes, so there is no need to open a new PR to make changes.